server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -739,7 +744,7 @@ protected final ShardSearchRequest buildShardSearchRequest(SearchShardIterator s
      */
     protected abstract SearchPhase getNextPhase();
 
-    private static final class PendingExecutions {
+    protected static final class PendingExecutions {
```

Comment:
this can stay private?

Reply:
Right, leftover from a more complicated earlier solution :)